extmod/uasyncio/event.py

```diff
@@ -29,3 +31,29 @@ async def wait(self):
             core.cur_task.data = self.waiting
             yield
         return True
+
+
+# MicroPython-extension: This can be set from outside the asyncio event loop,
+# such as other threads, IRQs or scheduler context. Implementation is a stream
+# that asyncio will poll until a flag is set.
+# Note: Unlike Event, this is self-clearing.
+try:
+    import uio
+
+    class ThreadSafeFlag(uio.IOBase):
+        def __init__(self):
+            self._flag = 0
+
+        def ioctl(self, req, flags):
+            if req == 3:  # MP_STREAM_POLL
+                return self._flag * flags
+            return None
+
+        def set(self):
+            self._flag = 1
+
+        async def wait(self):
+            yield core._io_queue.queue_read(self)
+            self._flag = 0
+except ImportError:
```

Comment:
2x blank lines needed here for black formatting

Reply:
Done